client/src/pages/trainer/learning-paths/create/index.tsx

```diff
@@ -47,7 +52,35 @@ const LearningPathCreate = (): JSX.Element => {
         }
         return await trigger(['name', 'description', 'category']);
       case 2:
-        return await push('/trainer/learning-paths');
+        try {
+          const formData = {
+            ...values,
+            category: values.category.map(({ id }) => id),
+            courses: values.courses.map(({ id, order }) => {
+              return {
+                course: id,
+                course_order: order,
+              };
+            }),
+            is_active: values.isActive,
+            image: typeof values.image === 'string' ? values.image : values.image?.name ?? null,
+          };
+
+          const res: any = await createLearningPath(formData);
+          if ('error' in res) {
+            const { data } = res.error;
+            const property = Object.keys(data)[0];
+            alertError(data[property][0]);
+          } else {
+            alertSuccess('Learning path created successfully!');
+            await push('/trainer/learning-paths');
+            dispatch(reset(null));
```

Comment:
may I know why you are passing `null` for reset  please ?

Reply:
I get a type error saying _"Expected 1 argument, but got 0"_

Fix: I removed the `action` argument from `features/learning-path/learningPathSlice`